### ISSUE TITLE
[New Resource]: AWS SES Mailmanager

### ISSUE THREAD

**Issue by Pacerino** [hashicorp/terraform-provider-aws#39008]
### Description

AWS SES has recently added a new function that allows you to manage incoming mails for SES or create rules and routes. This function is called “Mail Manager” and is mapped directly in SES, so it is not a separate new service. Right now without this

### Requested Resource(s) and/or Data Source(s)

aws_ses_relay
aws_ses_archive
aws_ses_ingress
aws_ses_traffic_policy

### References

https://github.com/aws/aws-sdk-go-v2/tree/main/service/mailmanager
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/mailmanager
https://docs.aws.amazon.com/ses/latest/dg/eb.html

### Would you like to implement a fix?

No

**Comment by justinretzolk** [comment]
Hey @Pacerino 👋 Thank you for taking the time to raise this! It looks like this falls under its own service in the AWS Go SDK, so it would wind up being a separate service in the AWS Provider as well (as in, the resources would be named something like `aws_mailmanager_*`. I've updated a couple of things about this issue with that in mind. There's no further action needed from you at this point, I just like to give people a heads up when I make those kinds of changes 🙂

**Comment by acwwat** [comment]
Adding a different perspective, the AWS CC Provider already has the resources and they are named `awscc_ses_mail_manager_*`, for example [`awscc_ses_mail_manager_relay`](https://registry.terraform.io/providers/hashicorp/awscc/latest/docs/resources/ses_mail_manager_relay). So I wonder if there is any merit following suit for the new resources to be created in the AWS Provider.

**Comment by dpalmerlumint** [comment]
Hey.  Will be doing a direct call to the CLI because these SES components aren't supported yet in Terraform.  Any idea on timeline?

**Comment by tzahari** [comment]
Any updates on this?

**Comment by GreasyAvocado** [comment]
Any news?

**Comment by paulgalow** [comment]
I agree that having official support in the AWS provider would be beneficial, despite what's already available with the AWSCC provider. 

I have implemented SES Mail Manager using the AWSCC provider and encountered several practical issues that native AWS provider support could resolve:

1. **Address List Members Not Supported**
   The [`awscc_ses_mail_manager_address_list`](https://registry.terraform.io/providers/hashicorp/awscc/latest/docs/resources/ses_mail_manager_address_list) resource creates the list but can't manage the email addresses themselves. So I currently add them after Terraform runs using direct API Calls or the AWS CLI.

2. **Tag Lifecycle Issues**
   Some AWSCC resources require `ignore_changes = [tags]` to prevent replacement on every apply (seen with ingress endpoints), breaking normal tag management workflows.

3. **Limited Data Source Support** 
   Unable to query existing SES Mail Manager resources within Terraform (had to use AWS provider's `aws_vpc_endpoint_service` data source for service discovery instead).

4. **Cloud Control API Limitations**
   AWSCC's Cloud Control API layer lacks the full feature set compared to native service APIs. For example: Resource deletions sometimes timeout due to Cloud Control API's polling mechanism, requiring manual intervention or state cleanup.

**Comment by GreasyAvocado** [comment]
I've already tried using AWSCC, and faced the same issues you described.
Full support as part of the standard AWS provider would help a lot.

**Comment by jessefarinacci** [comment]
> I have implemented SES Mail Manager using the AWSCC provider and encountered several practical issues that native AWS provider support could resolve:

Additionally, `awscc` is missing the truly awesome `aws` provider's multi-region support.

**Comment by sebastianhutter** [comment]
We'd like to add our voice to this feature request.

We are an AWS Partner managing infrastructure for numerous customers, all using Infrastructure as Code (Terraform). Several of our customers rely on AWS SES for email handling and specifically need Mail Manager for inbound email processing. In the eu-central-2 (Zurich) region, SES receipt rules are not available, Mail Manager is the only supported way to handle inbound mail routing and rules.

Currently, the only option is the AWSCC provider, which unfortunately has been unreliable in our experience. As a result, Mail Manager resources have to be managed manually, breaking the IaC workflow and causing disruptions in our CI/CD pipelines.

Native support for AWS Mailmanager in the AWS provider would allow us to fully manage these resources through Terraform, which is critical for maintaining consistent, automated infrastructure across our customer base.

Would love to see this prioritized. Happy to help with testing if needed.

**Comment by taruntej-a** [comment]
### AWS Mail Manager Resource Roadmap – Feedback Welcome

Hi everyone,
We're working on adding support for AWS Mail Manager resources and have identified the following implementation phases.

#### Phase 1 (Core Mail Manager Functionality)
We'll start with the resources that enable the core Mail Manager capabilities:

- `aws_mailmanager_ingress_point` - #49048 
- `aws_mailmanager_traffic_policy` - #49046 
- `aws_mailmanager_rule_set` - #49047

Our goal for Phase 1 is to provide the foundational resources needed for users to configure and operate the primary Mail Manager workflows.

#### Phase 2

Additional Mail Manager resources:

- `aws_mailmanager_relay` - #49049 
- `aws_mailmanager_archive` - #49050
- `aws_mailmanager_address_list` - #49051 
- `aws_mailmanager_addon_subscription` - #49052 
- `aws_mailmanager_addon_instance` - #49053 

#### Phase 3

Data Sources:

- `aws_mailmanager_traffic_policy` - #49054
- `aws_mailmanager_rule_set` - #49055
- `aws_mailmanager_address_list` - #49056

**Comment by taruntej-a** [comment]
We’re closing this issue since all Phase 1 items have been completed. Keeping this issue open won’t help us prioritize which resources to pick next, so please vote on the individual resource tickets using 👍.

We’ll still be picking up a few items from Phase 2 soon.

Thanks!
